public/app/features/plugins/plugin_loader.ts

```diff
@@ -218,3 +234,44 @@ export async function importAppPlugin(meta: PluginMeta): Promise<AppPlugin> {
 
   return plugin;
 }
+
+interface AddTranslationsToI18nOptions {
+  resolvedLanguage: string;
+  fallbackLanguage: string;
+  pluginId: string;
+  translations: Record<string, string>;
+}
+
+// exported for testing purposes only
+export async function addTranslationsToI18n({
+  resolvedLanguage,
+  fallbackLanguage,
+  pluginId,
+  translations,
+}: AddTranslationsToI18nOptions): Promise<void> {
+  const resolvedPath = translations[resolvedLanguage];
+  const fallbackPath = translations[fallbackLanguage];
+  const path = resolvedPath ?? fallbackPath;
+
+  if (!path) {
+    console.warn(`Could not find any translation for plugin ${pluginId}`, { resolvedLanguage, fallbackLanguage });
+    return;
+  }
+
+  try {
+    const module = await SystemJS.import(resolveModulePath(path));
+    if (!module.default) {
```

Comment:
In which scenario do we expect this to happen? Should we log a warning in this case?

Reply:
Good catch 🙌 , this should never happen but lets log a warning for this as well 👍

Reply:
Fixed 👍